src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -253,7 +257,12 @@ public string DisplayName
                     {
                         Span<char> flagsChars = stackalloc char[2];
                         HexConverter.ToCharsBuffer((byte)((~ActivityTraceFlagsIsSet) & _parentTraceFlags), flagsChars, 0, HexConverter.Casing.Lower);
-                        string parentId = "00-" + _traceId + "-" + _parentSpanId + "-" + flagsChars.ToString();
+                        string parentId =
+#if NET6_0_OR_GREATER
+                            string.Create(null, stackalloc char[128], $"00-{_traceId}-{_parentSpanId}-{flagsChars}");
```

Comment:
What is this magic overload

Reply:
See [String Interpolation in C# 10 and .NET 6](https://devblogs.microsoft.com/dotnet/string-interpolation-in-c-10-and-net-6/) and search for "Of course, we’re not encouraging everyone to author such a Create method on their own. That method is actually exposed on System.String in .NET 6"

Reply:
@jaredpar, now that we're planning to allow the compiler to do larger stack allocation implicitly for other features, we should revisit where we left off with string interpolation and whether the compiler can/should target this overload implicitly under certain circumstances.

Reply:
I agree we should discuss this in the near-ish future. This though basically requires the compiler to implicitly `stackalloc` for certain params under certain circumstances. Part of me wonders if we're too indirect with this parameter. Why not just change `DefaultInteroplationHandler` constructor itself to annotate the buffer parameter `[ImplicitStackAllocIfPossible]`? Not 100% sure that works (need to walk back through the uses and ref rules) but think it would.